research-software-directory/RSD-plugin-example
# Plugin01 example

# Example implementation of plugin-01 

This PR is using local RSD images and has their own docker-compose and nginx. Folder structure is changed to accommodate multiple plugins, custom rsd settings file, debug service and custom nginx. 

It includes Helmholtz PR.

Closes #1 

## How to test?

Use root readme file as guide. If not clear we should improve it. Check also README files in the specific "module" folders.

## Example user menu plugin 01
![image](https://github.com/user-attachments/assets/edd5bb44-88c6-467f-8979-df6cd75e30c8)

## Example edit software plugin 01
![image](https://github.com/user-attachments/assets/0c7f1e25-240c-4aa8-88c4-4c699054968a)

## Example plugin 01 frontend home page
![image](https://github.com/user-attachments/assets/d1c94d57-1e46-45aa-be60-40c7df60ffce)

## Example plugin 01 page linked by edit software link
![image](https://github.com/user-attachments/assets/0bc515e1-6bbe-4c86-8bf8-54e80cf14a12)

## Goal

The example repo should be starting point when developing new RSD plugin. The repo has docker-compose file and nginx configuration that integrates plugins and core RSD services. Using this approach the plugin developer should be able to test all functionality of the plugin and discover the limitations of current core RSD.  They can make changes in the core RSD services using the core repo.
